lib/theme_check/shopify_liquid/source_manager.rb

```diff
@@ -9,31 +15,68 @@ module SourceManager
 
       def download_or_refresh_files
         if has_required_files?
-          # TODO: https://github.com/Shopify/theme-check/issues/651
-          #
-          # Refresh files if they exist locally
+          refresh
         else
           download
         end
       end
 
-      def download
-        create_destination unless destination_exist?
+      def download(destination = documentation_directory)
+        create_destination(destination) unless destination_exist?(destination)
 
         required_file_names.each do |file_name|
-          download_file(local_path(file_name), remote_path(file_name))
+          download_file(local_path(file_name, destination), remote_path(file_name))
         end
       end
 
-      def local_path(file_name)
-        documentation_directory + "#{file_name}.json"
+      def refresh
+        return unless refresh_needed?
+
+        @refresh_thread = Thread.new do
+          tmp_dir = Dir.mktmpdir
```

Comment:
Personally, I'd pass a block (`Dir.mktmpdir { |tmp_dir| ... }`), so `tmpdir` takes care of removing the temporary directory for us :)